core/AssetManager/UIAsset/OnDiskUIAsset.php

```diff
@@ -58,12 +59,18 @@ public function delete()
     {
         if ($this->exists()) {
 
-            if (!unlink($this->getAbsoluteLocation()))
+            try {
+                Filesystem::remove($this->getAbsoluteLocation());
+            } catch (Exception $e) {
                 throw new Exception("Unable to delete merged file : " . $this->getAbsoluteLocation() . ". Please delete the file and refresh");
+            }
 
             // try to remove compressed version of the merged file.
-            @unlink($this->getAbsoluteLocation() . ".deflate");
-            @unlink($this->getAbsoluteLocation() . ".gz");
+            try {
+                Filesystem::remove($this->getAbsoluteLocation() . ".deflate");
+                Filesystem::remove($this->getAbsoluteLocation() . ".gz");
+            } catch (Exception $e) {
+            }
```

Comment:
Updated the code to optionally log a warning instead of throwing an exception. I've used a method parameter instead of defining a new method to avoid creating too many similar methods.